packages/components/src/components/SquiggleViewer/ViewerProvider.tsx

```diff
@@ -382,8 +382,11 @@ export const InnerViewerProvider = forwardRef<SquiggleViewerHandle, Props>(
     }, [playgroundSettings]);
 
     const handle: SquiggleViewerHandle = {
-      viewValuePath(path: SqValuePath) {
-        itemStore.scrollViewerToPath(path);
+      focusFromPath(path: SqValuePath) {
+        setZoomedInPathPath(undefined);
+        setTimeout(() => {
+          itemStore.focusFromPath(path);
+        }, 0);
```

Comment:
Probably not something you could fix right now, and I know this is not new (you used the same trick in keyboard navigation), but this bothers me.

React has time slicing, so in theory it could render things in multiple async passes, and then `setTimeout` won't be enough.

I can't think of a good way to delay the focus event, so I think we should consider going back to React-managed focus state (out of scope for this PR).

Reply:
Hm... good points on the time slicing, but I'd note that I am pretty nervous about going to React-managed focus. Maybe there's a decent hybrid we could do somehow.

Reply:
When I say "React-managed", I still mean "using the native focus, but sync in React -> DOM direction".

I assume it's possible to maintain the focus value by listening to `onBlur` and `onFocus` events, but will require some experimentation.

Another possible approach that I can see is to maintain the queue of focus events:
- create a ref in `ViewerProvider` for that queue, store it in context
- encapsulate "push to the queue" events through a custom hook, `const setFocus = useSetFocus()`, that appends to the queue
- then process the queue in `useEffect`; this also will give us the opportunity to resolve conflicts if we accidentally pushed more than one event on the queue

One problem with this is that sometimes you want to defer the focus and sometimes you don't (when you move focus from one item to another, `ViewerProvider` doesn't get rerendered). I hope that `setFocus(path, { defer: true })` would be ok here; if `defer` is false then it would apply immediately instead of pushing to the queue.

(ideally, the decision whether to defer should be done on the hook side, but I can't think of a good way to detect that)